.github/workflows/full-test.yml

```diff
@@ -120,97 +120,40 @@ jobs:
-          python ./.github/build_job_summary.py report_preprocessing.txt >> $GITHUB_STEP_SUMMARY  
-          rm report_preprocessing.txt
+        run: ./.github/run_tests.sh preprocessing
       - name: Test postprocessing
         if: ${{ steps.modules-changed.outputs.POSTPROCESSING_CHANGED == 'true' || steps.modules-changed.outputs.CORE_CHANGED == 'true' }}
-        run: |
-          source ${{ github.workspace }}/test_env/bin/activate
-          pytest -m postprocessing -vv -ra --durations=0 --durations-min=0.001 | tee report_postprocessing.txt; test ${PIPESTATUS[0]} -eq 0 || exit 1
-          echo "# Timing profile of postprocessing" >> $GITHUB_STEP_SUMMARY 
-          python ./.github/build_job_summary.py report_postprocessing.txt >> $GITHUB_STEP_SUMMARY  
-          rm report_postprocessing.txt
+        run: ./.github/run_tests.sh postprocessing
       - name: Test quality metrics
         if: ${{ steps.modules-changed.outputs.QUALITYMETRICS_CHANGED == 'true' || steps.modules-changed.outputs.CORE_CHANGED == 'true' }}
-        run: |
-          source ${{ github.workspace }}/test_env/bin/activate
-          pytest -m qualitymetrics -vv -ra --durations=0 --durations-min=0.001 | tee report_qualitymetrics.txt; test ${PIPESTATUS[0]} -eq 0 || exit 1
-          echo "# Timing profile of qualitymetrics" >> $GITHUB_STEP_SUMMARY 
-          python ./.github/build_job_summary.py report_qualitymetrics.txt >> $GITHUB_STEP_SUMMARY  
-          rm report_qualitymetrics.txt
+        run: ./.github/run_tests.sh qualitymetrics
       - name: Test core sorters
         if: ${{ steps.modules-changed.outputs.SORTERS_CHANGED == 'true' || steps.modules-changed.outputs.CORE_CHANGED == 'true' }}
-        run: |
-          source ~/test_env/bin/activate
-          pytest -m sorters -vv -ra --durations=0 --durations-min=0.001 | tee report_sorters.txt; test ${PIPESTATUS[0]} -eq 0 || exit 1
-          echo "# Timing profile of sorters" >> $GITHUB_STEP_SUMMARY 
-          python ./.github/build_job_summary.py report_sorters.txt >> $GITHUB_STEP_SUMMARY  
-          rm report_sorters.txt
+        run: ./.github/run_tests.sh sorters
       - name: Test comparison
         if: ${{ steps.modules-changed.outputs.COMPARISON_CHANGED == 'true' || steps.modules-changed.outputs.CORE_CHANGED == 'true' }}
-        run: |
-          source ~/test_env/bin/activate
-          pytest -m comparison -vv -ra --durations=0 --durations-min=0.001 | tee report_comparison.txt; test ${PIPESTATUS[0]} -eq 0 || exit 1
-          echo "# Timing profile of comparison" >> $GITHUB_STEP_SUMMARY 
-          python ./.github/build_job_summary.py report_comparison.txt >> $GITHUB_STEP_SUMMARY  
-          rm report_comparison.txt
+        run: ./.github/run_tests.sh comparison
       - name: Test curation
         if: ${{ steps.modules-changed.outputs.CURATION_CHANGED == 'true' || steps.modules-changed.outputs.CORE_CHANGED == 'true' }}
-        run: |
-          source ~/test_env/bin/activate
```

Comment:
This is what I forgot to change for `source ${{ github.workspace }}/test_env/bin/activate` @alejoe91 . Thats why #1531 is still failing.

Reply:
Let's fix it tomorrow! Thanks for looking into this :)

Reply:
Oh I see it's already fixed here. Will have a look and merge first thing in the morning